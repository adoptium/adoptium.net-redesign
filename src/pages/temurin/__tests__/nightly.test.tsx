import React from "react"
import { act, render, screen, fireEvent } from "@testing-library/react"
import { afterEach, describe, expect, it, vi } from "vitest"
import { axe } from "vitest-axe"
import Nightly, { Head } from "../nightly"
import AxiosInstance from "axios"
import MockAdapter from "axios-mock-adapter"

const mock = new MockAdapter(AxiosInstance)

afterEach(() => {
  vi.clearAllMocks()
})

describe("Temurin Nightly page", () => {
  it("renders correctly", async () => {
    mock.onGet().reply(200, [], { pagecount: 0 })

    const { container } = render(<Nightly />)
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")

    await act(async () => {
      const datepicker = screen.getByLabelText("Build Date")
      fireEvent.change(datepicker, { target: { value: "01/01/2022" } })
      expect(datepicker.getAttribute("value")).toBe("01/01/2022")
    })

    expect(pageContent).toMatchSnapshot()
  })

  it("head renders correctly", () => {
    mock.onGet().reply(200, [], { pagecount: 0 })

    const { container } = render(<Head />)
    // eslint-disable-next-line
    const title = container.querySelector("title")
    expect(title).toHaveTextContent("Nightly Builds | Adoptium")
  })

  it("has no accessibility violations", async () => {
    mock.onGet().reply(200, [], { pagecount: 0 });

    let container;
    await act(async () => {
      const renderResult = render(<Nightly />);
      container = renderResult.container;
    });

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
})
