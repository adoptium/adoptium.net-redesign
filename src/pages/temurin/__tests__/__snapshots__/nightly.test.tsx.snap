// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Temurin Nightly page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-10 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          Nightly builds
        </h1>
        <div
          class="row align-items-center pt-2"
        >
          <div
            class="callout callout-default text-start"
          >
            Text
            <br />
            <br />
            Text
            <br />
            “
            Text
            ”
            <br />
            <br />
            <p
              class="text-warning"
            >
              Text
            </p>
          </div>
          <div
            class="btn-group"
          >
            <a
              class="btn btn-primary m-3"
              href="/temurin/releases"
            >
              Text
               
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
                />
              </svg>
            </a>
            <a
              class="btn btn-secondary m-3"
              href="/temurin/archive"
            >
              Text
               
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <p
      class="text-center"
    >
      Text
    </p>
    <div
      class="input-group p-3 d-flex justify-content-center"
    >
      <label
        class="px-2 fw-bold"
        for="version"
      >
        Text
      </label>
      <select
        aria-label="version-filter"
        class="form-select form-select-sm"
        data-testid="version-filter"
        id="version-filter"
        style="max-width: 10em;"
      >
        <option
          value="2"
        >
          2 - EA
        </option>
        <option
          value="1"
        >
          1 - LTS
        </option>
        <option
          value="2"
        >
          2
        </option>
      </select>
    </div>
    <div
      class="input-group pb-5 d-flex justify-content-center"
    >
      <span
        class="p-2"
      >
        View
      </span>
      <select
        aria-label="Filter by number of builds"
        class="form-select form-select-sm"
        data-testid="build-num-filter"
        id="build-num-filter"
        style="max-width: 5em;"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          selected=""
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="50"
        >
          50
        </option>
      </select>
      <span
        class="p-2"
      >
        nightly builds prior to:
      </span>
      <div
        class="MuiFormControl-root MuiTextField-root css-qixv09-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r0:"
          id=":r0:-label"
        >
          Build Date
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            inputmode="text"
            placeholder="MM/DD/YYYY"
            type="text"
            value="01/01/2022"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date, selected date is Jan 1, 2022"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Build Date
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      id="nightly-list"
    >
      <table
        class="table table-hover text-start table-striped"
        id="nightly-table"
      >
        <thead
          class="table-dark"
        >
          <tr>
            <td
              class="fw-bold"
            >
              Platform
            </td>
            <td
              class="fw-bold"
            >
              Type
            </td>
            <td
              class="fw-bold"
            >
              Date
            </td>
            <td
              class="fw-bold"
            >
              Binary
            </td>
            <td
              class="fw-bold"
            >
              Installer
            </td>
            <td
              class="fw-bold"
            >
              SHA256
            </td>
          </tr>
        </thead>
        <tbody
          class="table-light"
        />
      </table>
    </div>
  </section>
</main>
`;
